docs/migrations/v3-to-v4.md

```diff
@@ -25,6 +25,25 @@ This is an API breaking change and as such IBC application developers will have
 
 ## IBC Light Clients
 
+### Light client and ClientState
+
 The `VerifyUpgradeAndUpdateState` function has been modified. The client state and consensus state return value has been removed. 
 
 Light clients **must** set the updated client state and consensus state in the client store after verifying a valid client upgrade.
+
+The `CheckHeaderAndUpdateState` function has been split into 4 new functions: `VerifyClientMessage`, `CheckForMisbehaviour`, `UpdateState`, 
+`UpdateStateOnMisbehaviour`
```

Comment:
```suggestion
The `CheckHeaderAndUpdateState` function has been split into 4 new functions: 
- `VerifyClientMessage` 
- `CheckForMisbehaviour` 
- `UpdateStateOnMisbehaviour`
- `UpdateState`, 

```

We should add a short description for each function. Maybe you could use the godoc from #1281?

Reply:
yesss